pkg/mock/verifications.go

```diff
@@ -0,0 +1,32 @@
+package mock
+
+// VerifyTimes defines how many times the MockServer may be called
+type VerifyTimes struct {
+	AtLeast int `json:"atLeast,omitempty"`
+	AtMost  int `json:"atMost,omitempty"`
+}
+
+// Verification defines how many times the MockServer may be called for a given request pattern
+type Verification struct {
+	Request *RequestMatcher `json:"httpRequest"`
+	Times   *VerifyTimes    `json:"times,omitempty"`
+}
+
+// CreateVerification creates a verification for a given expectation
+func CreateVerification(e *Expectation, vt *VerifyTimes) *Verification {
+	v := &Verification{
+		Request: e.Request,
+		Times:   vt,
+	}
+	return v
+}
+
+// CreateVerify creates a verification that bounds the number of times MockServer should have been called
+// Note: -1 is infinite number of times.
+func CreateVerify(min, max int) *VerifyTimes {
```

Comment:
It would be nice to have same approach with functional options for verifications as for expectations to make it uniform and also provide safe defaults (i.e. atMost: -1)